### PR TITLE
Move fil_actor_interface to forest shim

### LINKED ISSUE #4875: Move fil-actor-interface crate from fil-actor-states repo
# Summary

The [`fil_actor_interface`](https://github.com/ChainSafe/fil-actor-states/tree/main/fil_actor_interface) crate currently resides in the [`fil-actor-states`](https://github.com/ChainSafe/fil-actor-states) repo, but it should be moved into `forest` repo inside `shim/actors`.
This avoids the need to constantly re-release `fil_actor_interface` for minor changes that are only needed for us.
This crate is not being used by any external projects, so it makes sense to optimize the structure for internal use.

# Completion Criteria

- [ ] All references to `fil-actor-interface` within the forest repository are updated to reflect `shim/actors`.
- [ ] The `fil-actor-interface` crate is successfully moved from the `fil-actor-states` and it builds successfully with the moved `fil-actor-interface` crate.
- [ ] The `fil-actor-interface` crate is removed from the `fil-actor-states` repository to avoid confusion

# Additional Links & Resources

https://github.com/ChainSafe/fil-actor-states/blob/main/fil_actor_interface/Cargo.toml

### PR DESCRIPTION
**Pull request by sudo-shashank**
## Summary of changes

Changes introduced in this pull request:

- Move `fil_actor_interface` from [fil-actor-states](https://github.com/ChainSafe/fil-actor-states) to `shim/actors_interface`

## Reference issue to close (if applicable)

Closes #4875 

## Change checklist

- [X] I have performed a self-review of my own code,
- [X] I have made corresponding changes to the documentation. All new code adheres to the team's [documentation standards](https://github.com/ChainSafe/forest/wiki/Documentation-practices),
- [X] I have added tests that prove my fix is effective or that my feature works (if possible),
- [X] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -121,6 +122,7 @@ lru = "0.12"
 memmap2 = "0.9"
 memory-stats = "1"
 multiaddr = "0.18"
+multihash = { version = "0.18", features = ["identity"] }
```
Comment (hanabi1224) [inline Cargo.toml:125]: `multihash` is re-exported from `cid` and can be accessed by `cid::multihash`, let's not add it to the deps

File: src/fil_cns/validation.rs [hunk 2]
```diff
@@ -10,6 +10,7 @@ use crate::chain_sync::collect_errs;
 use crate::metrics::HistogramTimerExt;
 use crate::networks::{ChainConfig, Height};
 use crate::shim::actors::PowerActorStateLoad as _;
+use crate::shim::actors_interface::power;
```
Comment (hanabi1224) [inline src/fil_cns/validation.rs:13]: `_iterface` looks redundant when it's under `shim`, what do you think?

File: Cargo.toml (hunk 1 above)
Comment (sudo-shashank) [reply Cargo.toml]: fixed

File: src/fil_cns/validation.rs (hunk 2 above)
Comment (sudo-shashank) [reply src/fil_cns/validation.rs]: done

**Review by hanabi1224 — APPROVED** [review]
LGTM (But did not review the copy & pasted fil-actor-interface code line by line)

**Comment by sudo-shashank** [comment]
> LGTM (But did not review the copy & pasted fil-actor-interface code line by line)

those were copied verbatimly from `fil_actor_states` repo

**Comment by LesnyRumcajs** [comment]
@sudo-shashank is there a corresponding PR that removes this interface from `fil-actor-states`?

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**